tests/code/test_minor.py

```diff
@@ -132,26 +134,345 @@ def test_getCommunityEngagementByTerm():
+                                'username': 'khatts'}
+        
+        khattsEngagementData2 = {"id": 9,
+                                "matched": False, 
+                                "name": 'CELTS-Sponsored Event',
+                                'term': 3,
+                                "type": 'program',
+                                'username': 'khatts'}
+        
+        khattsEngagementData3 = {"id": 6,
+                                "matched": False, 
+                                "name": 'Habitat For Humanity',
+                                'term': 2,
+                                "type": 'program',
+                                'username': 'khatts'}
+        
+        khattsEngagementData4 = {"id": 4,
+                                "matched": False, 
+                                "name": 'People Who Care',
+                                'term': 3,
+                                "type": 'program',
+                                'username': 'khatts'}
+        
+        khattsEngagementData5 = {"id": 1,
+                                "matched": False, 
+                                "name": 'Databses',
+                                'term': 2,
+                                "type": 'course',
+                                'username': 'khatts'}
+        
+        neillzEngagementData1 = {"id": 4,
+                                "matched": False, 
+                                "name": 'People Who Care',
+                                'term': 3,
+                                "type": 'program',
+                                'username': 'neillz'}
+        
+        neillzEngagementData2 = {"id": 1,
+                                "matched": False, 
+                                "name": 'Databses',
+                                'term': 2,
+                                "type": 'course',
+                                'username': 'neillz'}
+        
+        setCommunityEngagementForUser('add', khattsEngagementData1, 'ramsayb2')
+        
+        khattsEngagements = IndividualRequirement.select()
+        # get count 
+        khattsEngagements.count() == 1
+        assert khattsEngagements[0].course == Course.get_by_id(2)
+        assert khattsEngagements[0].program == None
+
+        # add 4 more engagements and make sure the 5th raises the expected exception 
+        setCommunityEngagementForUser('add', khattsEngagementData2, 'ramsayb2')
+        setCommunityEngagementForUser('add', khattsEngagementData3, 'ramsayb2')
+        setCommunityEngagementForUser('add', khattsEngagementData4, 'ramsayb2')
+
+        with pytest.raises(DoesNotExist):
+            setCommunityEngagementForUser('add', khattsEngagementData5, 'ramsayb2')
+
+
+        # add records for another student and make sure it is added correctly. 
+        setCommunityEngagementForUser('add', neillzEngagementData1, 'ramsayb2')
+        neillzEngagements = IndividualRequirement.select()
```

Comment:
Going along with my previous comment, this variable name is not accurate since we are actually getting all IndividualRequirements. Not just Zach's.